### PR TITLE
Replace the overuse of StringView with std::string_view in Re2Function

### PR DESCRIPTION
**Pull request by xumingming**
Currently there are some overuse of StringView in Re2Functions.h/cpp. It's not good because StringView is not a replacement of std::string_view, its copy is expensive than std::string_view, its data() is not safe to point to for copied short StringView.

In this PR we replace StringView with std::string_view if it is a field of struct or a function param which is not feed by XxxVector directly.

### CODE REVIEW

**Comment by xumingming** [comment]
@mbasmanova

File: velox/functions/lib/Re2Functions.h [hunk 1]
```diff
@@ -118,7 +118,7 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> re2ExtractSignatures();
 /// characters} for patterns with wildcard characters only. Return
 /// {kGenericPattern, 0} for generic patterns).
 PatternMetadata determinePatternKind(
-    StringView pattern,
+    const std::string_view& pattern,
```
Comment (mbasmanova) [inline velox/functions/lib/Re2Functions.h:121]: Pass std::string_view by value.

File: velox/functions/lib/Re2Functions.cpp [hunk 2]
```diff
@@ -971,7 +967,7 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> re2ExtractSignatures() {
 }
 
 std::string unescape(
-    StringView pattern,
+    const std::string_view pattern,
```
Comment (mbasmanova) [inline velox/functions/lib/Re2Functions.cpp:970]: remove 'const'

**Comment by Yuhta** [comment]
Never mind it's pre-existing issue: https://github.com/facebookincubator/velox/issues/7856

**Pull request closed without merging**